Update dependencies, Docker images, and migrate from end-of-life Go 1.23

shatovilya opened the pull request:
# Description

Go 1.23 is no longer receiving security updates or support from the Go team. Refer to the end-of-life policy for Go at [endoflife.date/go.](https://endoflife.date/go)

Update dependencies and improve integration test stability:
- Update Go version from 1.23 to 1.24
- Update libsodium from 1.0.19 to 1.0.20
- Update libzmq from 4.3.4 to 4.3.5
- Update base image from debian11 to debian13
- Update Go dependencies

Fixes #416

## Type of change

Please select all options that apply to this change:

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [x] My code follows the style of this project.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding updates to the documentation.
- [x] I have added/updated unit tests to cover my changes.
- [x] I have added/updated integration tests to cover my changes.

Commit:
Update dependencies and improve integration test stability

- Update Go version from 1.23 to 1.24

- Update libsodium from 1.0.19 to 1.0.20

- Update libzmq from 4.3.4 to 4.3.5

- Update base image from debian11 to debian13

- Update Go dependencies to latest versions

Commit:
Chore: update Go version format in go.mod

Changed Go version from '1.24.0' to '1.24' to match convention.

agbpatro commented on .github/workflows/build.yml:
```diff
@@ -15,7 +15,7 @@ jobs:
       uses: actions/setup-go@v3
       with:
-        go-version: 1.23.0
+        go-version: 1.24.0
```
need to update [golangci-lint](https://github.com/teslamotors/fleet-telemetry/blob/aae5d8f39795995fe41d6e52ba1f31c2bfc51561/.github/workflows/build.yml#L46) as well

https://github.com/golangci/golangci-lint/releases

shatovilya replied:
@agbpatro 
Just wanted to check if this PR is still relevant and if there’s anything else I should update or adjust to help move it forward.

oderby commented:
Just pointing out that jumping several major versions for both golang and debian at the same time is somewhat risky (less so for golang than debian). I put up an open PR https://github.com/teslamotors/fleet-telemetry/pull/399 to go from debian 11 to 12 a while ago, which may be a more pragmatic approach to land that first, then proceed with upgrading golang. @agbpatro if there's interest, I'd be happy to get that PR rebased on latest master and ready to merge, just let me know.

Commit:
upgrade to Go 1.25 and update dependencies, tooling, and documentation

- Upgrade Go version to 1.25
- Update protoc from 4.25.1 to 29.0
- Update protoc-gen-go from v1.28 to v1.36.0
- Update all dependencies to compatible versions
- Regenerate protobuf files with new tool versions
- Fix reliable_ack to reliable_ack_sources in config documentation
- Add status_port, message_interval_time, and other missing config fields
- Add complete dispatcher configuration examples (pubsub, zmq, mqtt)
- Update dispatchers list with reliable ack support information
- Add airbrake and vins_signal_tracking_enabled config sections
- Update CI/CD pipeline configuration for Go 1.25

shatovilya committed:
Merge branch 'main' into update
